src/util/ringdelaybuffer.cpp

```diff
@@ -0,0 +1,211 @@
+        // The write position crossed the right bound of the buffer. So, if the situation,
+        // where the write position equals the read position arises, and before this situation,
+        // the read position does not cross the right bound of the buffer too,
+        // the buffer will be full.
+        m_fullFlag = true;
+    }
+
+    return itemsToWrite;
+}
+
+SINT RingDelayBuffer::moveReadPositionBy(const SINT jumpSize) {
+    if (jumpSize == 0) {
+        return 0;
+    }
+
+    // For the positive jumpSize values, the jump way is to right.
+    // For the negative jumpSize values, the jump way is to left.
+
+    // The number of available elements on the right side
+    // of the current read position.
+    const SINT readAvailableRight = getReadAvailable();
+
+    // The number of available elements on the left side
+    // of the current read position.
+    const SINT readAvailableLeft = -getWriteAvailable() + 1;
+
+    // Jump to the right is cannot be greater than the number
+    // of available items.
+    VERIFY_OR_DEBUG_ASSERT(jumpSize <= readAvailableRight) {
+        m_readPos = m_writePos;
+        // The buffer is empty (all elements have been read).
+        m_fullFlag = false;
+
+        return readAvailableRight;
+    }
+
+    // Jump to the left cannot be greater than the number of available items
+    // (in negative values smaller number).
+    VERIFY_OR_DEBUG_ASSERT(jumpSize >= readAvailableLeft) {
+        m_readPos = m_writePos;
+        // The buffer is full. In this situation, the values on the left side
+        // of the read position before the jump are considered as written,
+        // even though it is about default zero values. After the jump,
+        // the samples have to be read first before samples can be written.
+        m_fullFlag = true;
+
+        return readAvailableLeft;
+    }
+
+    // Add jump size to current read position.
+    m_readPos = m_readPos + jumpSize;
+
+    if (m_readPos >= m_buffer.size()) {
+        // Crossing the right bound of the buffer.
+        m_readPos = m_readPos - m_buffer.size();
+
+        // The read position crossed the right bound of the buffer. So, if the situation,
+        // where the read position equals the write position arises, and before this situation,
+        // the write position does not cross the right bound of the buffer too,
+        // the buffer will be empty.
+        m_fullFlag = false;
+    } else if (m_readPos < 0) {
+        // Crossing the left bound of the buffer.
+        m_readPos = m_readPos + m_buffer.size();
+
+        // The read position crossed the left bound of the buffer. So, if the situation,
+        // where the read position equals the write position arises, the buffer will be full.
+        // In this situation, the values on the left side of the read position before the jump
+        // are considered as written, even though it is about default zero values.
+        m_fullFlag = true;
```

Comment:
```suggestion
        m_fullFlag = m_readPos==m_writePos;
```

Reply:
Based on the new version proposed by @daschuer, this line of code was removed.